pkg/crd/crd.go

```diff
@@ -268,7 +268,11 @@ func (c *Client) ListPodIds(podns, podname string) (*[]aadpodid.AzureIdentity, e
 	var matchedIds []aadpodid.AzureIdentity
 	for _, v := range azAssignedIDList.(*aadpodid.AzureAssignedIdentityList).Items {
 		if v.Spec.Pod == podname && v.Spec.PodNamespace == podns {
-			matchedIds = append(matchedIds, *v.Spec.AzureIdentityRef)
+			for _, desiredState := range assignedIDStates {
+				if v.Status.Status == desiredState {
```

Comment:
strings.EqualFold ? or compareState function which we can change as per version ?